### PR TITLE
Feat/add edit

### PR DESCRIPTION
**Pull request by SHIKHASINGH1506**
1. add new note functionality implemented
2. edit existing note

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -1,9 +1,8 @@
 @import url('https://fonts.googleapis.com/css2?family=Montserrat:wght@400;500&family=Nunito:wght@200;400&display=swap');
 @import url('https://kube-ui.netlify.app/Components/kube-ui.css');
 @import url('https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.2/css/all.min.css');
-@import url('https://unpkg.com/mockman-js@latest/dist/style.css'); 
+/* @import url('https://unpkg.com/mockman-js@latest/dist/style.css');  */
```
Comment (bharati-21) [inline src/App.css:4]: I think you might have commented on this by mistake.
Dropping this so that you don't forget.

File: src/App.js [hunk 2]
```diff
@@ -2,33 +2,78 @@ import "./App.css";
...
+    }
+    setNoteData(initialNote);
+   addFormFocus 
+    ? dispatch({
+      type: 'SET_NEW_NOTE_FOCUS',
+      payload: {
+        addFormFocus: false,
+        editFormFocus: false
+      }
+      })
```
Comment (bharati-21) [inline src/App.js]: Nice implementation of outside click!

File: src/App.js [hunk 3]
```diff
@@ -2,33 +2,78 @@ import "./App.css";
...
+          autoClose={2000}
+          hideProgressBar={false}
+          newestOnTop={false}
+          closeOnClick
+          rtl={false}
+          draggable
+          pauseOnHover
+        />
+        <Navbar />
+        <Routes>
```
Comment (bharati-21) [inline src/App.js]: Suggestion - You could add these routes separately in a component. This way you can add private routes later on as well.

File: src/Pages/home-page/home.css [hunk 4]
```diff
@@ -1,3 +1,40 @@
 .wrapper {
   margin: 0 4.5rem;
 }
+.center-body {
+  width: 100%;
+  max-width: 100%;
+  /* position: relative; */
```
Comment (bharati-21) [inline src/Pages/home-page/home.css:7]: You can remove commented out code.

File: src/component/drawer/drawer.css [hunk 5]
```diff
@@ -32,23 +27,18 @@
   color: var(--primary-light-purple);
 }
 .drawer-app-content {
-  position: relative;
+  /* position: relative; */
```
Comment (bharati-21) [inline src/component/drawer/drawer.css:30]: Again, you can remove this.

File: src/component/search-bar/search-bar.css [hunk 6]
```diff
@@ -1,6 +1,6 @@
 .search-bar {
   flex-grow: 1;
-  position: relative;
+  /* position: relative; */
```
Comment (bharati-21) [inline src/component/search-bar/search-bar.css:3]: Here as wel.

File: src/service/note-service.js [hunk 7]
```diff
@@ -0,0 +1,75 @@
+import axios from 'axios';
+
+const getNotes = async (dispatch) => {
+  const token = JSON.parse(localStorage.getItem("token"));
+  const Headers = { authorization: token };
+  try {
+    const { data :{notes}, status } = await axios.get('/api/notes', { headers: Headers });
+    if(status>=200 && status<=300){
```
Comment (bharati-21) [inline src/service/note-service.js:8]: You don't need to check the status if you use Axios. You can directly use catch block to handle calls that were not successful (>=200 && < 300)

File: src/App.js [hunk 8]
```diff
@@ -2,33 +2,78 @@ import "./App.css";
 import 'react-toastify/dist/ReactToastify.css';
 
 import { Routes, Route } from "react-router-dom";
-import {ToastContainer} from 'react-toastify';
-import Mockman from 'mockman-js';
+import { ToastContainer } from 'react-toastify';
+// import Mockman from 'mockman-js';
```
Comment (bharati-21) [inline src/App.js:4]: You can remove the commented out code.

File: src/App.js [hunk 9]
```diff
@@ -2,33 +2,78 @@ import "./App.css";
...
+          <Route path='/' element={<LandingPage />} />
+          {/* <Route path='/mockman' element={<Mockman />} /> */}
+          <Route path='/home' element={<Home />} />
+          <Route path='/login' element={<Login />} />
+          <Route path='/signup' element={<Signup />} />
+        </Routes>
+      </div>
     </div>
   );
 }
```
Comment (bharati-21) [inline src/App.js:25]: One suggestion I have is that if it's possible you can move the onPageClickHandler to a separate file. Doing so will make your App component look cleaner.

**Review by bharati-21 — APPROVED** [review]
I really liked your implementation of outside note click! It's very clever.

I only had a few suggestions and you can choose to work on whenever you have the bandwidth.

File: src/service/note-service.js (hunk 7 above)
Comment (SHIKHASINGH1506) [reply src/service/note-service.js:8]: But what if we want to show specific error to user, if someone is unauthorized to particular content then we need to filter 401 status and if there's internal server error then 500. 
I've not implemented checking for rest of the status code here but may add later. In that case is it fine to use?

File: src/App.js (hunk 9 above)
Comment (SHIKHASINGH1506) [reply src/App.js:25]: I'll refactor this.

**Pull request merged**